### DISCUSSION TITLE
Manual/UIE-get-started-with-runtime-ui

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#546] [Notes]
# Manual/UIE-get-started-with-runtime-ui

https://docs.unity3d.com/Manual/UIE-get-started-with-runtime-ui.html

**Comment by vertxxyz** [comment]
To support Live Reload (toggled via right-clicking on an editor tab and selecting *UI Toolkit Live Reload*) you should use `OnEnable` to set up your UI, and `OnDisable` to tear it down. This is because Live Reload will call those methods on all components present on the reloading UIDocument.
If you aren't often performing this logic and are only supporting it for Live Reload, I recommend clearing any caches, removing elements from the hierarchy and rebuilding the UI from scratch to avoid ongoing state issues.
